File: go/number_guessing_game/number_guessing_game_9.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

// This program is a number guessing game where the user has to guess a randomly generated number between 1 and 100.
// The program provides feedback on whether the guess is too high, too low, or correct.

func main() {
    rand.Seed(time.Now().UnixNano())
    target := rand.Intn(100) + 1
    var guess int

    fmt.Println("Welcome to the Number Guessing Game!")
    fmt.Println("I have selected a random number between 1 and 100.")
    fmt.Println("Can you guess what it is?")

    for {
        fmt.Print("Enter your guess: ")
        fmt.Scan(&guess)

        if guess < target {
            fmt.Println("Too low!")
        } else if guess > target {
            fmt.Println("Too high!")
        } else {
            fmt.Println("Congratulations! You guessed the number!")
            break
        }
    }

    fmt.Println("Thank you for playing the Number Guessing Game!")
}

File: vbnet/number_guessing_game/number_guessing_game_9.vb
' This program is a number guessing game written in Visual Basic .NET.
' The purpose of the game is to guess a randomly generated number between 1 and 100.
' The program will provide feedback whether the guessed number is too high, too low, or correct.

Module NumberGuessingGame
    Sub Main()
        Dim random As New Random()
        Dim targetNumber As Integer = random.Next(1, 101)
        Dim guess As Integer = 0

        Console.WriteLine("Welcome to the Number Guessing Game!")
        Console.WriteLine("Try to guess the number between 1 and 100.")

        While guess <> targetNumber
            Console.Write("Enter your guess: ")
            guess = Convert.ToInt32(Console.ReadLine())

            If guess < targetNumber Then
                Console.WriteLine("Too low! Try again.")
            ElseIf guess > targetNumber Then
                Console.WriteLine("Too high! Try again.")
            Else
                Console.WriteLine("Congratulations! You guessed the correct number.")
            End If
        End While

        Console.WriteLine("Thank you for playing the Number Guessing Game!")
    End Sub
End Module

